src/smallmap_gui.cpp

```diff
@@ -1004,6 +1005,43 @@ class SmallMapWindow : public Window {
 		}
 	}
 
+	/**
+	 * Adds industry names to the smallmap.
+	 * @param dpi the part of the smallmap to be drawn into
+	 */
+	void DrawIndustryNames(const DrawPixelInfo *dpi) const
+	{
+		if (this->map_type != SMT_INDUSTRY) return;
+
+		for (const Industry *i : Industry::Iterate()) {
+			const LegendAndColour &tbl = _legend_from_industries[_industry_to_list_pos[i->type]];
+			if (!tbl.show_on_map) continue;
```

Comment:
Bit inefficient to iterate thru all industries before checking the legend `show_on_map` flag first.

However, I couldn't find something like this: `Industry::IterateByType(IndustryType)`

```
for (int i = 0; i != _smallmap_industry_count; i++) {
	const LegendAndColour &tbl = _legend_from_industries[i];
	for (const Industry *ind : Industry::IterateByType(tbl.type)) {
		…
	}
}
```